src/Symfony/Bundle/TwigBundle/DependencyInjection/Compiler/ExceptionListenerPass.php

```diff
@@ -39,13 +39,11 @@ public function process(ContainerBuilder $container)
 
         if ($container->hasParameter('templating.engines')) {
             $engines = $container->getParameter('templating.engines');
-            if (\in_array('twig', $engines, true)) {
-                $container->removeDefinition('exception_listener');
-
-                return;
+            if (!\in_array('twig', $engines, true)) {
+                $container->removeDefinition('twig.exception_listener');
             }
         }
 
-        $container->removeDefinition('twig.exception_listener');
+        $container->removeDefinition('exception_listener');
```

Comment:
If you really want to revert, the code should be set back as is before #38350.
Of there is a difference with before #38350, can you please highlight it?

Reply:
Well, I'm going to try to describe my process step by step (it's easier for me since I'm not fluent):

- first, I had to rollback before #38350 was merged
- then I recovered the tests brought by #38350
- then I ran them (and `testRemovesTwigExceptionListenerIfTwigIsNotUsedAsTemplateEngine` failed due to the behavior changes before/after #38350)
- then I decided to cover the current behavior of the bundle (at this state, which is still 4.4.14)
- then they passed (excepted for the one which was marked as skipped, which covered the change of behavior)
- then I decided to manually applied the changes of #38350
- then I notice that my new made test failed

So I decided to refactor a bit the code, since anyway its was now covered by the tests.
Is it more clear ? 

Should I still revert completely the code ?

Thanks

Reply:
The behaviour before #38350 was that the `exception_listener` definition was removed if any templating engine was active. So probably no exception logging would happen in the application!

The proposed change of @adaniloff would change the conditions when the `exception_listener` is killed to:
Precondition: `twig.exception_listener.controller` is not `null`

1. The templating engine is active with the twig template engine (indicated by parameter `templating.engines`) 
2. The templating engine is not active

The second conditions looks wrong for me. Why should the twig exception listener should be preferred over the default exception listener for that case?

Reply:
@dmolineus The thing is that I understand why you did #38350 and I do not disagree with this change :) !  However it caused the issue related to this MR (https://github.com/symfony/symfony/issues/38482). 

However, the more I think about it, the more I think this is related to some error prone configuration.

It may be caused by some "under-the-hood" issue that #38350 finally reveals.
I'm going to try to see if this is true (or not) and get back to you.

Reply:
@adaniloff Regarding to the docs the second parameter has to be a `\Symfony\Component\Debug\Exception\FlattenException`:

> Your controller will be passed two parameters: exception, which is a \Symfony\Component\Debug\Exception\FlattenException instance created from the exception being handled, and logger, an instance of \Symfony\Component\HttpKernel\Log\DebugLoggerInterface (which may be null).

https://symfony.com/doc/2.5/cookbook/controller/error_pages.html#replacing-the-default-exceptioncontroller

Also the default twig error listener expects a flatten exception. https://github.com/symfony/symfony/blob/6000e5162901f02be2da7f3ffc41717b41b2d2e5/src/Symfony/Bundle/TwigBundle/Controller/ExceptionController.php#L59

So I guess the rest bundle also have to deal with it which it doesn't right now:  
https://github.com/FriendsOfSymfony/FOSRestBundle/blob/3d8501dbdfa48811ef942f5f93c358c80d5ad8eb/Controller/ExceptionController.php#L65

Reply:
Yep, before submitting the issue, I had already gone through all of these thoughts. The issue for me is, even if the behavior <  #38350 (so < 4.4.15) is incorrect, having some drastic change like the one from 4.4.15 make me doubt about the "patch" increment. 

Should'nt it be more like a 4.5.0 ? I get it, it's a bugfix, however the bug seems to be pretty old so... is it not a little radical to fix it that way ?

Just asking, I'm not sure myself.